src/topology/algebra/ordered.lean

```diff
@@ -1632,6 +1632,54 @@ nhds_within_Iio_ne_bot (le_refl a)
 
 end linear_order
 
+section decidable_linear_order
+
+variables [topological_space α] [decidable_linear_order α] [order_topology α] [densely_ordered α]
+
+/-- The `at_top` filter for an open interval `Ioo a b` comes from the left-neighbourhoods filter at
+the right endpoint in the ambient order. -/
+lemma Ioo_at_top_eq_nhds_within {a b : α} (h : a < b) :
+  at_top = comap (coe : Ioo a b → α) (𝓝[Iio b] b) :=
+begin
+  haveI : nonempty (Ioo a b) := ⟨classical.choice (nonempty.to_subtype (dense h))⟩,
```

Comment:
This seems like it should be a lemma (if it isn't already) -- you should just be able to write `haveI := Ioo.nonempty h` or something. I see the `classical.choice ...` mouthful three times in this PR alone!

Reply:
Good idea!  I added lemmas (instances in some cases) for nonemptyness of all the interval types.  Also, the `classical.choice` was only needed one of the 3 times in this PR, the rest was a silly oversight (now fixed) where I switched from nonempty to inhabited and then back.